include/libtorrent/i2p_stream.hpp

```diff
@@ -193,7 +193,7 @@ struct i2p_stream : proxy_base
 
 		// send hello command
 		m_state = read_hello_response;
-		static const char cmd[] = "HELLO VERSION MIN=3.0 MAX=3.0\n";
+		static const char cmd[] = "HELLO VERSION MIN=3.1 MAX=3.1\n";
```

Comment:
is there a new feature this update depend on that's only available since SAM version 3.1?

Reply:
https://geti2p.net/en/docs/api/samv3:
_SAM 3.1 is the recommended minimum SAM implementation because of its support for better signature types than SAM 3.0._
I guess @zzzi2p can explain better why 3.1 is needed (#7258).

Reply:
3.1 has been supported since 2014 and is the only version supported by i2pd. 3.1 gives you SIGNATURE_TYPE=7 (Ed25519) which gets your users off of insecure DSA_SHA1 signatures. There is no risk of finding a pre-2014 router to talk to, they wouldn't work anyway.